pkg/collector/python/kubeutil.go

```diff
@@ -34,9 +35,12 @@ var (
 func getConnections() map[string]string {
 	kubeutil, err := kubelet.GetKubeUtil()
 	if err != nil {
-		// Connection to the kubelet fail, return empty dict
+		// Connection to the kubelet fail, return the error
 		log.Errorf("connection to kubelet failed: %v", err)
-		return nil
+		if e, ok := err.(*retry.Error); ok {
```

Comment:
We could probably use `errors.As` here? https://golang.org/pkg/errors/#example_As

Reply:
Good point !
https://github.com/DataDog/datadog-agent/pull/6315/commits/0a3af81224cb89e13882c4e490a2f12e628d397c